File: app/src/main/java/com/luigi573/mira/statistics/StatisticsModule.java
package com.luigi573.mira.statistics;

import com.luigi573.mira.di.ActivityScoped;
import com.luigi573.mira.di.FragmentScoped;

import dagger.Binds;
import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class StatisticsModule {
    @FragmentScoped
    @ContributesAndroidInjector
    abstract StatisticsFragment statisticsFragment();

    @Binds
    @ActivityScoped
    abstract StatisticsContract.Presenter statisticsPresenter(StatisticsPresenter presenter);
}

File: app/src/main/java/com/luigi573/mira/statistics/StatisticsModule.kt
package com.luigi573.mira.statistics

import androidx.lifecycle.ViewModel
import com.luigi573.mira.di.ViewModelBuilder
import com.luigi573.mira.di.ViewModelKey
import dagger.Binds
import dagger.Module
import dagger.android.ContributesAndroidInjector
import dagger.multibindings.IntoMap

@Module
abstract class StatisticsModule {
    @ContributesAndroidInjector(modules = [
        ViewModelBuilder::class
    ])
    abstract fun statisticsFragment(): StatisticsFragment

    @Binds
    @IntoMap
    @ViewModelKey(StatisticsViewModel::class)
    abstract fun bindViewModel(viewModel: StatisticsViewModel): ViewModel
}